curriculum/challenges/english/10-coding-interview-prep/data-structures/remove-an-element-from-a-max-heap.md

```diff
@@ -95,16 +110,23 @@ assert(
     } else {
       return false;
     }
-    test.insert(30);
-    test.insert(300);
-    test.insert(500);
-    test.insert(10);
-    let result = [];
-    result.push(test.remove());
-    result.push(test.remove());
-    result.push(test.remove());
-    result.push(test.remove());
-    return result.join('') == '5003003010';
+ let arr;
+ let max = Infinity;
+  const [result, vals] = [[], [2, 15, 3, 7, 12, 7, 10, 90]];
+  vals.forEach((val) => test.insert(val));
+  for (let i = 0; i < vals.length; i++) {
+    const removed = test.remove();
+    if(removed > max) return false
+    max = removed;
+    result.push(removed);
+    test.print()[0] === null
+      ? (arr = test.print().slice(1))
+      : (arr = test.print());
```

Comment:
```suggestion
    const curHeap = test.print();
    const arr = curHeap[0] === null ? curHeap.slice(1) : curHeap;
```
Suggestion for defining `arr` within loop, and slightly clearer assigning to it.